lib/portlayer/storage/vsphere/volume.go

```diff
@@ -87,6 +87,28 @@ func (v *VolumeStore) AddStore(ctx context.Context, ds *datastore.Helper, storeN
 	return u, nil
 }
 
+func (v *VolumeStore) VolumeStoresList(ctx context.Context) (map[string]url.URL, error) {
+	m := make(map[string]url.URL)
+	for u, ds := range v.ds {
+
+		// Get the ds:// URL given the datastore url ("[datastore] /path")
+		dsurl, err := datastore.ToURL(ds.RootURL)
```

Comment:
@matthewavery are you concerned about a pointer?  ds is a map  of the following type

```
39 »···// maps datastore uri (volume store) to datastore
40 »···ds map[url.URL]*datastore.Helper 
```Copying of a pointer is trivial unless youre concerned about something else.  

Also, about your proposed solution;  in general, avoid iterating maps whenever possible. But if that can't be avoided (like here), don't then use the result of the iteration to once again look the iterated key up in the map.